### ISSUE TITLE
hyriseBenchmarkJoinOrder --verify fails

### ISSUE THREAD

**Issue by mrzzzrm** [hyrise/hyrise#1409]
Update: SQLite is working since #1419. Now we got different problems, see comments

**Comment by mrzzzrm** [comment]
After #1419, some queries get successfully verified. Some (e.g. 10a) do not finish within reasonable time (left running 3h), some report mismatches (e.g. 7a). Will continue investigating

**Comment by mrks** [comment]
> Interestingly 10a runs almost instantly when issued from the sqlite command prompt.

How did you load the data here?

**Comment by mrzzzrm** [comment]
The JOB has a schema.sql - after running that I probably just loaded the imdb data as normal csv into sqlite

**Comment by mrks** [comment]
That schema.sql uses `PRIMARY KEY`, so you can join on an index.

**Comment by mrks** [comment]
Are JOB queries supposed to return empty results (e.g., 12b)? While sqlite and Hyrise agree, we might want to check if the data is loaded correctly.

**Comment by mrzzzrm** [comment]
Until now, I have kind of lived with some queries returning empty results... Unprofessional as it is. Since the queries are about movies/actors etc, it should be possible to figure out whether there should be matches or not.

**Comment by mrks** [comment]
If loading the data into sqlite / mysql also results in empty results, everything is fine. No need to actually look at the queries.

**Comment by mrzzzrm** [comment]
I think - but this is a year ago! - that running the queries via the sqlite3 cli also produced the occasional empty result set. But we should re-verify this before we forever shrug off these resultless queries as "that's just the way it is"

**Comment by mrks** [comment]
Current results so that you don't have to wait an hour: https://gist.github.com/mrks/3b28c649e328e87078bde4db8cb5ecfd

**Comment by mrks** [comment]
Any chance the original export into CSV files lost something?

```
$ wc -l ~/imdb_data/movie_info_idx.csv
1380035 /home/Markus.Dreseler/imdb_data/movie_info_idx.csv
$ wc -l movie_info_idx.csv
2367725 movie_info_idx.csv
```

However, [following the online guide](https://github.com/gregrahn/join-order-benchmark), I could not get it to finish without problems:

```
# TIME createIndexes() : 7min, 15sec (wall) 0min, 0sec (user) 0min, 0sec (system)
adding foreign keys (this may take a while)
ERROR caught exception creating a foreign key: insert or update on table "aka_title" violates foreign key constraint "movie_id_exists"
DETAIL:  Key (movie_id)=(0) is not present in table "title".

# TIME createForeignKeys() : 1min, 44sec (wall) 0min, 0sec (user) 0min, 0sec (system)
```

Also, some tables are smaller than the ones you generated.

**Comment by mrzzzrm** [comment]
I thought I commented here already... It is certainly possible I did something wrong during the table generation. I think I used sqlite instead of the default postgres for imdbpy.

**Comment by mrzzzrm** [comment]
I will not fix the --verify failure as part of #1577 - JoinHash gets tested quite heavilty there and --verify still fails. I think it is probably rather related to TableScans on String columns, but that is speculation. I think looking at this is a different block of work than #1577 and we should not block #1577 because of it.

**Comment by mrks** [comment]
ok
